File: Add Code Here/C++/leetcode/Letter Combinations of a Phone Number.cpp
class Solution {
public:
    vector<string> answer;
    void solver(vector<string> numstr,string ans,string digit,int index){
       if(index==digit.size()){
           answer.push_back(ans);
           return;
       }
        // cout<<numstr[digit[index]]<<endl;
        int n=digit[index]-'0';
        for(int i=0;i<numstr[n].size();i++){
            solver(numstr,ans+numstr[n][i],digit,index+1);
        }
    }
    vector<string> letterCombinations(string digits) {
        if(digits.size()==0) return answer;
        vector<string> numstr={"","","abc","def","ghi","jkl","mno","pqrs","tuv","wxyz"};
        string ans="";
        solver(numstr,ans,digits,0);
        return answer;
    }
};

File: Add Code Here/Java/LetterCombinationsOfaPhoneNumber.java
/* Given a string containing digits from 2-9 inclusive, return all possible letter combinations that the number could represent. Return the answer in any order.

A mapping of digits to letters (just like on the telephone buttons) is given below. Note that 1 does not map to any letters. */



class Solution {
    public List<String> letterCombinations(String digits) {
        LinkedList<String> ans = new LinkedList<String>();
		if(digits.isEmpty()) return ans;
		String[] mapping = new String[] {"0", "1", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"};
		ans.add("");
        for(int i = 0; i < digits.length(); i++){
            int z = Character.getNumericValue(digits.charAt(i));
            while(ans.peek().length() == i){
                String pontu = ans.remove();
                for(char c : mapping[z].toCharArray()){
                    ans.add(pontu + c);
                }
            } 
        }
        return ans;
    }
}